internal/core/data/v2/application/event.go

```diff
@@ -73,41 +69,21 @@ func AddEvent(e models.Event, profileName string, deviceName string, ctx context
 	return nil
 }
 
-// PublishEvent publishes incoming AddEventRequest through MessageClient
-func PublishEvent(addEventReq dto.AddEventRequest, profileName string, deviceName string, ctx context.Context, dic *di.Container) {
+// PublishEvent publishes incoming AddEventRequest in the format of []byte through MessageClient
+func PublishEvent(data []byte, profileName string, deviceName string, sourceName string, ctx context.Context, dic *di.Container) {
 	lc := container.LoggingClientFrom(dic.Get)
 	msgClient := dataContainer.MessagingClientFrom(dic.Get)
 	configuration := dataContainer.ConfigurationFrom(dic.Get)
 	correlationId := correlation.FromContext(ctx)
 
-	lc.Debug("Putting V2 Event DTO on message queue", clients.CorrelationHeader, correlationId)
-
-	var data []byte
-	var err error
-
-	if len(clients.FromContext(ctx, clients.ContentType)) == 0 {
-		ctx = context.WithValue(ctx, clients.ContentType, clients.ContentTypeJSON)
-	}
-
-	// Must make sure API Version for embedded DTOs is set since it isn't required by the request,
-	// but is needed when published to Message Bus.
-	addEventReq.Event.Versionable = common.NewVersionable()
-	for index := range addEventReq.Event.Readings {
-		addEventReq.Event.Readings[index].Versionable = common.NewVersionable()
-	}
-
-	data, err = json.Marshal(addEventReq)
-	if err != nil {
-		lc.Error(fmt.Sprintf("error marshaling V2 AddEventRequest DTO: %+v", addEventReq), clients.CorrelationHeader, correlationId)
-		return
-	}
+	lc.Debug("Publishing V2 AddEventRequest to message queue", clients.CorrelationHeader, correlationId)
```

Comment:
Please add the full topic the event is published to, i.e. `publishTopic` below.

Reply:
updated https://github.com/edgexfoundry/edgex-go/pull/3202/commits/a97e7d088cfea97e2f7e544ceb56995fb0b64fad